api.planx.uk/saveAndReturn/resumeApplication.test.ts

```diff
@@ -4,31 +4,33 @@ import app from "../server";
 import { queryMock } from "../tests/graphqlQueryMock";
 import { mockLowcalSession, mockTeam } from "../tests/mocks/saveAndReturnMocks";
 import { buildContentFromSessions } from "./resumeApplication";
+import { PartialDeep } from "type-fest";
 
 const ENDPOINT = "/resume-application";
 const TEST_EMAIL = "simulate-delivered@notifications.service.gov.uk";
 
-type DeepPartial<T> = T extends object
-  ? {
-      [P in keyof T]?: DeepPartial<T[P]>;
-    }
-  : T;
-
 const mockFormatRawProjectTypes = jest
   .fn()
   .mockResolvedValue(["New office premises"]);
 
 jest.mock("@opensystemslab/planx-core", () => {
+  const actualCoreDomainClient = jest.requireActual(
+    "@opensystemslab/planx-core",
+  ).CoreDomainClient;
+
   return {
-    CoreDomainClient: jest.fn().mockImplementation(() => ({
-      formatRawProjectTypes: () => mockFormatRawProjectTypes(),
-    })),
+    CoreDomainClient: class extends actualCoreDomainClient {
+      constructor() {
+        super();
+        this.formatRawProjectTypes = () => mockFormatRawProjectTypes();
+      }
+    },
   };
 });
```

Comment:
This structure is repeated a few times throughout this PR. I spent a good bit of time attempting to generalise it so that we could import it with an interface similar to - 

```ts
function mockCoreDomainClient(customMocks = {}) {
   const actualCoreDomainClient = jest.requireActual(
    "@opensystemslab/planx-core",
  ).CoreDomainClient;

  return {
    CoreDomainClient: class extends actualCoreDomainClient {
      constructor() {
        super();
        Object.assign(this, customMocks)
      }
    }
  }
}

...

mockCoreDomainClient({
  formatRawProjectTypes: () => mockFormatRawProjectTypes(),
});
```

Tried a few variations without much success - I suspect the problem is related to how Jest is hoisting mocks (through the `jest.mock("@opensystemslab/planx-core")` notation). I wanted to just move on as opposed to losing more time here so I'll open a GitHub issue describing this.

Reply:
This structure is repeated a few times throughout this PR. I spent a good bit of time attempting to generalise it so that we could import it with an interface similar to - 

```ts
function mockCoreDomainClient(customMocks = {}) {
   const actualCoreDomainClient = jest.requireActual(
    "@opensystemslab/planx-core",
  ).CoreDomainClient;

  return {
    CoreDomainClient: class extends actualCoreDomainClient {
      constructor() {
        super();
        Object.assign(this, customMocks)
      }
    }
  }
}

...

mockCoreDomainClient({
  formatRawProjectTypes: () => mockFormatRawProjectTypes(),
});
```

Tried a few variations without much success - I suspect the problem is related to how Jest is hoisting mocks (through the `jest.mock("@opensystemslab/planx-core")` notation). I wanted to just move on as opposed to losing more time here so I'll open a GitHub issue describing this.

Reply:
Issue logged here - https://github.com/theopensystemslab/planx-new/issues/2404